crates/turbopack-core/src/resolve/mod.rs

```diff
@@ -1257,10 +1299,58 @@ fn handle_exports_field(
     // other options do not apply anymore when an exports field exist
     Ok(merge_results_with_references(
         resolved_results,
-        vec![AffectingResolvingAssetReferenceVc::new(package_json).into()],
+        vec![AffectingResolvingAssetReferenceVc::new(package_json_path).into()],
     ))
 }
 
+/// Resolves a `#dep` import using the containing package.json's `imports`
+/// field. The dep may be a constant string or a pattern, and the values can be
+/// static strings or conditions like `import` or `require` to handle ESM/CJS
+/// with differently compiled files.
+async fn resolve_package_internal_with_imports_field(
+    context: FileSystemPathVc,
+    request: RequestVc,
+    resolve_options: ResolveOptionsVc,
+    pattern: &Pattern,
+    conditions: &BTreeMap<String, ConditionValue>,
+    unspecified_conditions: &ConditionValue,
+) -> Result<ResolveResultVc> {
+    let Pattern::Constant(specifier) = pattern else {
+        bail!("PackageInternal requests can only be Constant strings");
```

Comment:
Why is this argument a `pattern` and not a string? Can this actually occur at runtime?